### PR TITLE
Feature/620 multiple transport coverage

### PR DESCRIPTION
**Pull request by ismaelbej**

### CODE REVIEW

File: cpp/walking-coverage.cpp [hunk 1]
```diff
@@ -964,15 +994,30 @@ int main(int argc, char *argv[])
...
+    unique_ptr<float[]> new_cost =
+      run_dijkstra_on_friction_layer(friction.get(),
+                                     width, height, nodata,
+                                     frictionRaster.pixel_width_meters(),
+                                     frictionRaster.pixel_height_meters(),
+                                     pixelOrigin.first, pixelOrigin.second,
+                                     options._maxTimeCost[i],
+                                     options._minFriction[i]);
+
+    // Merging layers scale costs to make it is easier to calculate the isochrone
```
Comment (ggiraldez) [inline cpp/walking-coverage.cpp:1018]: This comment is a bit misleading. I'd say you need to adjust the new cost layer to take it to the same scale as the original in order to compute the isochrone.

File: client/src/planwise/client/coverage.cljs [hunk 2]
```diff
@@ -81,15 +81,20 @@
 
 (defn coverage-algorithm-filter-options
   [{:keys [coverage-algorithm value on-change empty disabled?]}]
-  (let [list      (rf/subscribe [:coverage/supported-algorithms])
-        criteria  (get-in @list [(keyword coverage-algorithm) :criteria])]
+  (let [list         (rf/subscribe [:coverage/supported-algorithms])
+        criteria     (get-in @list [(keyword coverage-algorithm) :criteria])
+        valid-keys   (keys criteria)
+        update-value (fn [key change]
+                       (if (= coverage-algorithm "drive-walk-friction")
```
Comment (ggiraldez) [inline client/src/planwise/client/coverage.cljs:88]: I don't think you need this `if` here (plus the special case handling should is a smell). In fact, the previous behavior was wrong in that it only allowed a single criterion. Filtering the keys (using `select-keys`) is the correct and general solution.

File: client/src/planwise/client/projects2/components/settings.cljs [hunk 3]
```diff
@@ -232,20 +232,31 @@
      [:label "Tags: " [tag-set @tags read-only]]
      [count-providers @tags @current-project provider-unit]]))
 
+; Accept previous coverage algorithms as the new one combining both
+(defn- get-coverage-algorithm
+  [coverage]
+  (case coverage
+    "driving-friction" "drive-walk-friction"
+    "walking-friction" "drive-walk-friction"
+    coverage))
```
Comment (ggiraldez) [inline client/src/planwise/client/projects2/components/settings.cljs:241]: I think we should do this migration server side given that the list of supported algorithms now only returns `drive-walk-friction`, eg. in `planwise.component.projects2/get-project`. You _could_ probably remove the old cases of `compute-coverage-polygon` in that case as well.

**Review by ggiraldez — COMMENTED** [review]
This looks great! But, please do check the comments, especially regarding the keys to the default `frictions` var.

File: src/planwise/component/coverage.clj [hunk 4]
```diff
@@ -79,21 +80,48 @@
   (let [db-spec         (:spec db)
         friction-raster (friction/find-friction-raster db-spec coords)
         max-time        (:walking-time criteria)
-        min-friction    (float (/ 1 100))]
+        min-friction    (:walking-friction frictions)]
```
[the remark this answers is not in the document]
Comment (ismaelbej) [reply src/planwise/component/coverage.clj:83]: I'll fix it.

File: src/planwise/component/coverage.clj [hunk 5]
```diff
@@ -79,21 +80,48 @@
...
+                                 :friction [min-friction]})
       (throw (ex-info "Cannot find a friction raster for the given coordinates" {:coords coords})))))
 
 (defmethod compute-coverage-polygon :driving-friction
   [{:keys [db runner]} coords criteria]
   (let [db-spec         (:spec db)
         friction-raster (friction/find-friction-raster db-spec coords)
         max-time        (:driving-time criteria)
-        min-friction    (float (/ 1 2000))]
+        min-friction    (:driving-friction frictions)]
```
[the remark this answers is not in the document]
Comment (ismaelbej) [reply src/planwise/component/coverage.clj:97]: I'll fix it.

File: cpp/walking-coverage.cpp (hunk 1 above)
Comment (ismaelbej) [reply cpp/walking-coverage.cpp:1018]: I'll write a better explanation.

File: client/src/planwise/client/coverage.cljs (hunk 2 above)
Comment (ismaelbej) [reply client/src/planwise/client/coverage.cljs:88]: I'll fix it.

File: client/src/planwise/client/projects2/components/settings.cljs (hunk 3 above)
Comment (ismaelbej) [reply client/src/planwise/client/projects2/components/settings.cljs:241]: I'll do it.

**Pull request merged**